packages/server/test/utils/xdccDatabase.test.ts

```diff
@@ -1,79 +1,219 @@
 
 			await create(parsedXdccDatabase);
-			const result = await search("foo");
 
-			expect(result).toStrictEqual([
-				{
-					channelName: "AeC|1P|01",
-					fileName: "Foo-Fuugther.part01.rar",
-					fileNumber: "#1",
-					fileSize: "1.9G",
-					network: "irc.foo.biz",
-					serverName: "#a-b-c",
-				},
+			expect(mockSqlite3.Database).toHaveBeenCalledWith(":memory:");
+			expect(mockPrepare).toHaveBeenCalledWith("INSERT INTO files VALUES (?, ?, ?, ?, ?, ?)");
+		});
+
+		it("should filter out invalid script lines", async () => {
+			mockFetch.mockResolvedValue({
+				text: () =>
+					Promise.resolve(`#1   Bot1 100M Valid File.txt
+invalid line
+#2
+   
+#3   Bot2 200M Another Valid.txt`),
+			} as Response);
+
+			await create([
 				{
-					channelName: "AeC|1P|01",
-					fileName: "Foo-Fuugther.part02.rar",
-					fileNumber: "#2",
-					fileSize: "1.9G",
-					network: "irc.foo.biz",
-					serverName: "#a-b-c",
+					channelName: "#test",
+					network: "irc.test.net",
+					scriptUrl: "http://test.com/script.php",
 				},
+			]);
+
+			expect(mockPrepare).toHaveBeenCalled();
+		});
+	});
+
+	describe("search", () => {
+		it("should search by filename", async () => {
+			const testResults = [
 				{
-					channelName: "AeC|1P|01",
-					fileName: "Foo-Fuugther.part01.rar",
+					channelName: "#test-channel",
+					network: "irc.test.net",
 					fileNumber: "#1",
-					fileSize: "1.9G",
-					network: "irc.foo.biz",
-					serverName: "#Pierpaolo",
-				},
-				{
-					channelName: "AeC|1P|01",
-					fileName: "Foo-Fuugther.part02.rar",
-					fileNumber: "#2",
-					fileSize: "1.9G",
-					network: "irc.foo.biz",
-					serverName: "#Pierpaolo",
+					botName: "TestBot",
+					fileSize: "500M",
+					fileName: "Some File Name.txt",
 				},
-			]);
+			];
+
+			mockAll.mockImplementation((_query, _params, callback) => {
+				callback(null, testResults);
+			});
+
+			const result = await search("some name");
+
+			expect(result).toHaveLength(1);
+			expect(result[0]).toMatchObject({
+				fileName: "Some File Name.txt",
+				id: "irc.test.net-#test-channel-TestBot-#1-Some File Name.txt-500M",
+			});
+		});
+
+		it("should handle empty search results", async () => {
+			mockAll.mockImplementation((_query, _params, callback) => {
+				callback(null, []);
+			});
+
+			const result = await search("nonexistent");
+
+			expect(Array.isArray(result)).toBe(true);
+			expect(result).toHaveLength(0);
+		});
+
+		it("should convert search terms to LIKE pattern", async () => {
+			mockAll.mockImplementation((_query, params, callback) => {
+				expect(params[0]).toBe("%test%file%");
+				callback(null, []);
+			});
+
+			await search("test file");
+		});
+
+		it("should handle undefined rows gracefully", async () => {
+			mockAll.mockImplementation((_query, _params, callback) => {
+				callback(null, undefined as unknown as unknown[]);
+			});
+
+			const result = await search("test");
+
+			expect(Array.isArray(result)).toBe(true);
```

Comment:
check for real values, instead of just checking if that's an array

apply the same for all the other tests